### ISSUE TITLE
Request: Params (ParamArrayAttribute) replace with ...

### ISSUE THREAD

**Issue by thisbarker** [yanghuan/CSharp.lua#331]
Would it be possible to add a conversion of the final method parameter if it includes the attribute `ParamArrayAttribute` to be converted to `...`? I believe this is the equivalent in Lua from C# per my understanding.

**Comment by yanghuan** [comment]
is you want invoke a function from c code like printf?

**Comment by thisbarker** [comment]
print(...) output
```
value 1 2 another value 3 4
```

Example Lua to replicate with c#
```
function Test(...)
  print(...)
end

Test("value 1", 2, "another value 3", 4);
```

Expected C# to replicate Lua
```
public void Test(params object[] args)
  Console.WriteLine(args);
end

Test("value 1", 2, "another value 3", 4);
```

**Comment by yanghuan** [comment]
you can use like this 
https://github.com/yanghuan/CSharp.lua/wiki/attributes_en#csharpluatemplate

```
 /// <summary>
 /// @CSharpLua.Template = "print({*0})"
 /// </summary>
 public extern static void Print(params object[] args);
```

**Comment by Cheatoid** [comment]
Right. That's already possible but *only* when defining API.
What about the other way around?

----

@yanghuan I think @thisbarker wants to have `...` defined on the generated Lua function from actual C# function.
So if I am right, this is ***not*** about defining C# API (with `Template` comment), but rather an actual C# function that is to be transpiled to Lua function *should have* `...` inserted to its parameters, did I understood you correctly @thisbarker ?

So, what if you write a C# function which you want to have `...` (vararg parameter) inserted on the Lua function?
For example, C# **user** code:
```cs
static void Foo(string bar, ????) { /* actual C# code */ }
```
In other words, how do you tell *CSharp.Lua* to insert `...` on the generated Lua function `Foo`?
Whereas the **expected** generated Lua function code would be:
```lua
Foo = function(bar, ...)
--                  ^^^ vararg "..." added
end
```

----

May I suggest, either of the following approaches,
- Use `__arglist` (the _hidden_ keyword):
  `static void Foo(string bar, __arglist) { /* actual C# code */ }`

or,
- Introduce a new attribute "`VarArg`" and use `params object[] args`:
  `static void Foo(string bar, [VarArg] params object[] args) { /* actual C# code */ }`

**Comment by thisbarker** [comment]
> (Also, you can not have `[ParamArray]` on the `params` parameter.)

@Cheatoid it is automatically added when you prefix the parameter with `params`
